server/routerlicious/packages/services-shared/src/socketIoServer.ts

```diff
@@ -146,9 +146,21 @@ class SocketIoServer implements core.IWebSocketServer {
 		private readonly apiCounters: IApiCounters = new InMemoryApiCounters(),
 	) {
 		this.io.on("connection", (socket: Socket) => {
+			/**
+			 * Fluid Socket.io connection URL looks like:
+			 * "<hostname>/socket.io/?documentId=[documentId]&tenantId=[tenantId]&EIO=[3/4]&transport=[websocket/polling]"
+			 * [socket.handshake.query](https://socket.io/docs/v4/server-socket-instance/#sockethandshake) contains parsed query params.
+			 * The following properties are used for **telemetry purposes only.**
+			 * These should **not** be used to identify the tenant and document associated with the socket connection
+			 * for real logic and access purposes without validating against the JWT access token.
```

Comment:
Is this the only way to add these properties? Wondering about the issue you raised in the comment here

Reply:
It's not the "only way," but it's the only way at this level. The JWT isn't sent until there is a `connect_document` message, so to validate these telemetry properties we'd have to wait until that message is sent and alter this metric from a lower level (where it's consumed by Nexus).

Do you have particular concerns? IMO for telemetry, this is fine. Technically, a malicious user could mess up our telemetry by using an invalid tenant or document, but that would also show up as 403 errors with weird tenants or docs in our telemetry for the same connections.